mdata/notifierKafka/notifierKafka.go

```diff
@@ -69,22 +69,77 @@ func (c *NotifierKafka) start() {
 	pre := time.Now()
 	processBacklog := new(sync.WaitGroup)
 	for _, partition := range partitions {
-		var offsetTime int64
+		type offset struct {
+			offsetStart int64
+			offsetTime  int64
+			offsetName  string
+			offsetError error
+		}
+		var startOffset int64
+		var validOffset bool
+		var offsetFromDuration int64
+
+		offsets := make([]offset, 0, 3)
```

Comment:
I think this would be easier to follow by using a map[string]offset, keyed by the offsetName.
eg
```
offsets := map[string]offset{
    "oldest": {offsetTime: sarama.OffsetOldest},
    "newest": {offsetTime: sarama.OffsetNewest},
    "custom": {offsetTime: offsetFromDuration},
}
```

To iterate over them in order you can use
```
for _, name := range []string{"oldest", "newest", "custom"} {
    tmpOffset, err := c.client.GetOffset(topic, partition, offsets[name].offsetTime)
    offsets[name].offsetStart = tmpOffset
    offsets[name].offsetError = err
}
```